Core/src/Detector/VolumeStructureBuilder.cpp

```diff
@@ -0,0 +1,188 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Detector/VolumeStructureBuilder.hpp"
+
+#include "Acts/Detector/PortalGenerators.hpp"
+#include "Acts/Geometry/ConeVolumeBounds.hpp"
+#include "Acts/Geometry/CuboidVolumeBounds.hpp"
+#include "Acts/Geometry/CutoutCylinderVolumeBounds.hpp"
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Geometry/GenericCuboidVolumeBounds.hpp"
+#include "Acts/Geometry/TrapezoidVolumeBounds.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+
+namespace {
+
+/// This can be abandoned with C++20 to use the std::to_array method
+///
+/// @note only the first kDIM elments will obviously be filled, if the
+/// vector tends to be longer, it is truncated
+///
+/// @param vecvals the vector of bound values to be converted
+/// @return an array with the filled values
+template <std::size_t kDIM>
+std::array<Acts::ActsScalar, kDIM> to_array(
+    const std::vector<Acts::ActsScalar>& vecvals) {
+  std::array<Acts::ActsScalar, kDIM> rarray = {};
+  for (const auto [iv, v] : Acts::enumerate(vecvals)) {
+    if (iv < kDIM) {
+      rarray[iv] = v;
+    }
+  }
+  return rarray;
+}
```

Comment:
since we have documentation on this now and it looks kind of generic - should we move this to a header in utilities?

Reply:
Can do, no problem.